File: sdk/atlas_verify.py
"""Atlas verify — a tiny, dependency-free Python client for the verified-human endpoint.

Drop this one file into your app; it uses only the standard library. See docs/VERIFY_QUICKSTART.md.

    from atlas_verify import is_live_human
    if is_live_human("https://node.clockwork-tree.com", token, min_tier=2):
        ...  # a live human, tier >= 2, verified offline by signature
"""
from __future__ import annotations

import json
import urllib.request
from typing import Any, Dict, Optional


def verify_presence(node_url: str, token: str, *, fmt: str = "vc", audience: Optional[str] = None,
                    min_tier: int = 1, public_key: Optional[str] = None, timeout: float = 10.0) -> Dict[str, Any]:
    """POST a presented Atlas credential to `/verify/presence` and return the verdict dict:
    {valid, verified_human, assurance_tier, epoch, subject}. `public_key` is optional — the node
    resolves the presenter from the token's DID if the persona registered (see /did/register)."""
    body: Dict[str, Any] = {"format": fmt, "token": token, "min_tier": min_tier}
    if audience is not None:
        body["audience"] = audience
    if public_key is not None:
        body["public_key"] = public_key
    req = urllib.request.Request(node_url.rstrip("/") + "/verify/presence",
                                 data=json.dumps(body).encode(),
                                 headers={"Content-Type": "application/json"}, method="POST")
    with urllib.request.urlopen(req, timeout=timeout) as r:
        return json.loads(r.read().decode())


def is_live_human(node_url: str, token: str, *, min_tier: int = 1, **kw: Any) -> bool:
    """True iff the token proves a live human at assurance tier >= `min_tier`."""
    return bool(verify_presence(node_url, token, min_tier=min_tier, **kw).get("valid"))

File: sdk/atlas-verify.js
// Atlas verify — a tiny, dependency-free JS client for the verified-human endpoint.
// Node 18+ (built-in fetch) or any modern browser. See docs/VERIFY_QUICKSTART.md.
//
//   import { isLiveHuman } from "./atlas-verify.js";
//   if (await isLiveHuman("https://node.clockwork-tree.com", token, { minTier: 2 })) {
//     // a live human, tier >= 2, verified offline by signature
//   }

/**
 * POST a presented Atlas credential to /verify/presence and return the verdict object:
 * { valid, verified_human, assurance_tier, epoch, subject }.
 * `publicKey` is optional — the node resolves the presenter from the token's DID if registered.
 */
export async function verifyPresence(nodeUrl, token, { format = "vc", audience, minTier = 1, publicKey } = {}) {
  const body = { format, token, min_tier: minTier };
  if (audience != null) body.audience = audience;
  if (publicKey != null) body.public_key = publicKey;
  const res = await fetch(nodeUrl.replace(/\/+$/, "") + "/verify/presence", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  });
  if (!res.ok) throw new Error(`verify failed: HTTP ${res.status}`);
  return res.json();
}

/** True iff the token proves a live human at assurance tier >= minTier. */
export async function isLiveHuman(nodeUrl, token, opts = {}) {
  return Boolean((await verifyPresence(nodeUrl, token, opts)).valid);
}
